pyproject.toml

```diff
@@ -5,7 +5,8 @@ requires = ["maturin>=1.2.0,<1.3.0"]
 [project]
 authors = [{name = "Eventual Inc", email = "daft@eventualcomputing.com"}]
 dependencies = [
-  "pyarrow >= 6.0.1",
+  "pyarrow >= 12.0.0; platform_system != 'Windows'",
```

Comment:
we should leave the pin here at 6.0.1, we have some users are using pyarrow 7.0.0 or pyarrow 8.0.0